mobile_app_connector/controllers/mobile_app_controller.py

```diff
@@ -124,19 +125,18 @@ def get_hero(self, view=None, **parameters):
         res = hero_mapping.get_connect_data(hero)
         return [res]
 
-    @http.route('/sponsor_a_child/<string:lang_code>/<string:source>',
+    @http.route('/sponsor_a_child',
                 type='http', auth='public', website=True)
-    def mobile_app_sponsorship_request(self, lang_code=None, source=None,
-                                       partner_id=False, **parameters):
+    def mobile_app_sponsorship_request(self, **parameters):
         """
         Create a sms_child_request and redirect user to sms sponsorship form
         It uses sms sponsorship
         :return: Redirect to sms_sponsorship form
         """
         values = {
-            'lang_code': lang_code,
-            'source': source,
-            'partner_id': partner_id,
+            'lang_code': _get_lang(parameters),
+            'source': parameters['source'],
+            'partner_id': parameters.get('partner_id', False),
```

Comment:
```suggestion
            'partner_id': parameters.get('partner_id'),
```
It's already returning False by default if the parameter is absent.